src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12404,6 +12363,50 @@ private SyntaxList<SyntaxToken> ParseAnonymousFunctionModifiers()
             return result;
         }
 
+        private bool IsAnonymousFunctionAsyncModifier()
+        {
+            Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.AsyncKeyword);
+
+            switch (this.PeekToken(1).Kind)
+            {
+                case SyntaxKind.OpenParenToken:
+                case SyntaxKind.IdentifierToken:
+                case SyntaxKind.StaticKeyword:
+                case SyntaxKind.RefKeyword:
+                case SyntaxKind.DelegateKeyword:
+                    return true;
+                case var kind:
+                    return IsPredefinedType(kind);
```

Comment:
does this work for `async void (...`

also, this is likely a good test to have if we don't have it already.

Reply:
> does this work for `async void (...`

Yes, this handle `void`. There are a number of tests in `LambdaReturnTypeParsingTests` for `void` returning lambdas with and without modifiers.